### PR TITLE
Find and fix bug in test_vcache_flush and library

### PR DESCRIPTION
**Pull request by save-buffer**

### CODE REVIEW

**Comment by drichmond** [comment]
Looks good, but I don't understand what the bug was. Can you describe?

**Comment by save-buffer** [comment]
Essentially, the mask doesn't do anything for reads and only affects writes. When you send a read packet requesting a word, the manycore will send the entire word, regardless of what the mask is. As a result, you have to shift it down. 
Similarly, when you send a write packet, you don't send the word you want in the low-order bits, you have to shift that up to the proper position along with the mask.

**Comment by drichmond** [comment]
@mrutt92 LGTY?

File: libraries/bsg_manycore.cpp [hunk 1]
```diff
@@ -1363,22 +1363,22 @@ static int hb_mc_manycore_recv_read_rsp(hb_mc_manycore_t *mc,
         }
 
         /* check that the npa matches? */
-
+        
+        int shift = CHAR_BIT * (hb_mc_npa_get_epa(npa) & 0x3);
```
Comment (mrutt92) [inline libraries/bsg_manycore.cpp]: Technically CHAR_BIT is the number of bits in a byte on the host machine ;-)

Comment (drichmond) [reply libraries/bsg_manycore.cpp]: So, sizeof(char)*8?

Comment (mrutt92) [reply libraries/bsg_manycore.cpp]: Nah, it's fine as is! I'm just giving a hard time.

**Pull request merged**